google/resource_container_node_pool_test.go

```diff
@@ -390,6 +405,29 @@ func testAccCheckContainerNodePoolDestroy(s *terraform.State) error {
 	return nil
 }
 
+func TestAccContainerNodePool_regionalClusters(t *testing.T) {
```

Comment:
This is unusual, but this resource is unusually complicated - could you also find a way to test that updates work correctly?  Maybe add another similar config and a test step to update from this config to that one?

Reply:
I am also concerned about nested node pools - I see that you have a plan to write an acceptance test for those tomorrow, which is great.

Reply:
What do you mean by unusual? It felt pretty complicated to me too.

Reply:
It's not an important note - I'm just saying that this resource contains more twists and turns than most - most resources have very clean "create, update, delete" cycles where each step makes exactly one (simple) request, but the nature of these container node pools means that they have lots of special cases.

Reply:
Oh!  Sorry, to clarify: my *request* is unusual - normally we would be very happy with this test as it stands, but since this resource is so much more complicated, I'd like to see some other testing.

Reply:
Ah I see! Yes I'm fine to add more tests :-) 

Two more cases make sense to me:
- Doing an update (if this is possible)
- nested node pool

Thoughts?